integration-tests/src/bin/gen_blockchain_data.rs

```diff
@@ -211,6 +230,53 @@ async fn main() {
         (block_num.as_u64(), contract.address()),
     );
 
+    // Deploy smart contracts for worst case block benches
+    //
+
+    // CheckMload
+    let contract = deploy(
```

Comment:
I think here there is a clear opportunity for DRY. There is always deploy, get_block_number, blocks.insert, deployments.insert. So I think it can be extracted into a function

Reply:
@leolara, correct. I was thinking about it and trying to figure out if i can extract the constructor arguments' type from the compiled contract somehow, cause it is an input to the .deploy() method, and differs depending on the .sol code. So for example, MLOAD.sol does not have a constructor, so deploy() needs () as input. Maybe work on this separately and put it in another PR to not broaden the scope so much? @ChihChengLiang ?

Reply:
It looks like the refactor some additional work. I think putting the deduplication in the next pull request is a good idea.

Reply:
OK